test/SparqlAntlrParserTest.cpp

```diff
@@ -1863,6 +1858,59 @@ TEST(SparqlParser, updateQueryUnsupported) {
   expectUpdateFails("COPY GRAPH <a> TO GRAPH <a>", updateUnsupported);
 }
 
+TEST(SparqlParser, Quads) {
+  auto expectQuads = ExpectCompleteParse<&Parser::quads>{defaultPrefixMap};
+  auto expectQuadsFails = ExpectParseFails<&Parser::quads>{};
+  auto Iri = [](std::string_view stringWithBrackets) {
+    return TripleComponent::Iri::fromIriref(stringWithBrackets);
+  };
+
+  expectQuads("?a <b> <c>",
+              UnorderedElementsAre(m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"),
+                                           std::monostate{})));
+  expectQuads("GRAPH <foo> { ?a <b> <c> }",
+              UnorderedElementsAre(
+                  m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"), ::Iri("<foo>"))));
+  expectQuads("GRAPH <foo> { ?a <b> <c> } GRAPH <bar> { <d> <e> ?f }",
+              UnorderedElementsAre(
+                  m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"), ::Iri("<foo>")),
+                  m::Quad(Iri("<d>"), Iri("<e>"), Var("?f"), ::Iri("<bar>"))));
+  expectQuads(
+      "GRAPH <foo> { ?a <b> <c> } . <d> <e> <f> . <g> <h> <i> ",
+      UnorderedElementsAre(
+          m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"), ::Iri("<foo>")),
+          m::Quad(Iri("<d>"), Iri("<e>"), Iri("<f>"), std::monostate{}),
+          m::Quad(Iri("<g>"), Iri("<h>"), Iri("<i>"), std::monostate{})));
+  expectQuads(
+      "GRAPH <foo> { ?a <b> <c> } . <d> <e> <f> . <g> <h> <i> GRAPH <bar> { "
+      "<j> <k> <l> }",
+      UnorderedElementsAre(
+          m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"), ::Iri("<foo>")),
+          m::Quad(Iri("<d>"), Iri("<e>"), Iri("<f>"), std::monostate{}),
+          m::Quad(Iri("<g>"), Iri("<h>"), Iri("<i>"), std::monostate{}),
+          m::Quad(Iri("<j>"), Iri("<k>"), Iri("<l>"), ::Iri("<bar>"))));
+
+  expectQuads(
+      "GRAPH <foo> { ?a <b> <c> } . <d> <e> <f> . <g> <h> <i> . GRAPH <bar> { "
+      "<j> <k> <l> }",
+      UnorderedElementsAre(
+          m::Quad(Var("?a"), Iri("<b>"), Iri("<c>"), ::Iri("<foo>")),
+          m::Quad(Iri("<d>"), Iri("<e>"), Iri("<f>"), std::monostate{}),
+          m::Quad(Iri("<g>"), Iri("<h>"), Iri("<i>"), std::monostate{}),
+          m::Quad(Iri("<j>"), Iri("<k>"), Iri("<l>"), ::Iri("<bar>"))));
+}
+
+TEST(SparqlParser, QuadData) {
+  auto expectQuadData =
+      ExpectCompleteParse<&Parser::quadData>{defaultPrefixMap};
+  auto expectQuadDataFails = ExpectParseFails<&Parser::quadData>{};
+
+  expectQuadDataFails("<a> <b> ?c");
+  expectQuadDataFails("<a> <b> <c> . GRAPH <foo> { <d> ?e <f> }");
+  expectQuadDataFails("<a> <b> <c> . ?d <e> <f> }");
+  expectQuadDataFails("GRAPH ?foo { <a> <b> <c> }");
```

Comment:
Here the case of succesfull parsing seems to be missing (e.g. `<a> <b> <c>` if I understand correctly).